jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java

```diff
@@ -174,6 +174,9 @@ public ProcessInstance getProcessInstance(long id, boolean readOnly) {
                 ((ProcessInstanceImpl) processInstance).reconnect();
                 if (readOnly) {
```

Comment:
I am starting to think if we need to make all this changes ... maybe we could simply not reconnect the process instance - not registering any listeners.... as this is just for reading the state of the process instance and should never react to any singals...wdyt?

Reply:
+1 from me, actually I think we discussed this approach but I wasn't sure if it would break something else or not. Then this would mean to just add else branch for `if (readOnly) {` with the code `((ProcessInstanceImpl) processInstance).reconnect();` I think.

Reply:
it does not work.

Reply:
@MarianMacik it is not enough to reconnect... the listeners can be added anywhere so you have no control where they are added. not all the logic is centralized so it is rather difficult to unravel the whole thing.